candy-machine/program/src/processor/mint.rs

```diff
@@ -229,17 +235,30 @@ pub fn handle_mint_nft<'info>(
     if let Some(gatekeeper) = &candy_machine.data.gatekeeper {
         if ctx.remaining_accounts.len() <= remaining_accounts_counter {
             punish_bots(
-                CandyError::GatewayTokenMissing,
+                Some(CandyError::GatewayTokenMissing),
                 payer.to_account_info(),
                 ctx.accounts.candy_machine.to_account_info(),
                 ctx.accounts.system_program.to_account_info(),
                 BOT_FEE,
+                None,
             )?;
             return Ok(());
         }
         let gateway_token_info = &ctx.remaining_accounts[remaining_accounts_counter];
+
         remaining_accounts_counter += 1;
 
+        let gateway_verification_result = {
```

Comment:
I added this block today and from the output, its saying that the programId is invalid from gateway_token_info. Could also explain why there is no data being created on the account creation. The PDA has been cross checked with the JS version and it is correct. The programId being passed in from the test is also correct as far I can tell.